### PR TITLE
Create Windows Build

### PR DESCRIPTION
**Pull request by bitoffdev**
2021-01-31_windows_tetris

## Changelog

- Create build instructions for windows using MinGW
- Make event library cross-platform and use it for requests
- Add -f flag for logging to file

### CODE REVIEW

File: docs/WINDOWS.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+Change to the directory that you would like to contain the source code.
+
+For the rest of this guide, we will assume you are using your "Documents"
+directory.
+
+## 4. Clone and Build PDCurses
+
+```powershell
+git.exe clone https://github.com/wmcbrine/PDCurses
```
Comment (Nitepone) [inline docs/WINDOWS.md:25]: I'd prefer this to be a submodule as we do with Unity.

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -1,10 +1,21 @@
 cmake_minimum_required (VERSION 3.3)
 
+# Previously, we used add_definitions() to set the compile flags, but that is not always supported.
+set(CMAKE_C_STANDARD 99)
+
+if (WIN32)
```
Comment (Nitepone) [inline test/CMakeLists.txt:6]: I assume this is defined by cmake? Can we build for Windows builds from an Ubuntu env? Quick Google search points me at mingw-w64 for cross platform builds.

File: src/tetris_game.c [hunk 3]
```diff
@@ -9,6 +9,7 @@
 #include <string.h>
 #include <time.h>
 
+#include "os_compat.h"
```
Comment (Nitepone) [inline src/tetris_game.c:12]: Curious, why?

File: docs/WINDOWS.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+make
+```
+
+## 6. [Optional] Run Tests
+
+```powershell
+.\bin\unit_tests.exe
+```
+
+*Note: Some tests (that rely on a specific Linux `rand_r` seeding) will fail on
```
Comment (Nitepone) [inline docs/WINDOWS.md:71]: :(

**Review by Nitepone — APPROVED** [review]

File: src/tetris_game.c (hunk 3 above)
Comment (bitoffdev) [reply src/tetris_game.c:12]: `tetris_game.c` uses `rand_r`, which is defined in `os_compat.h` for Windows

File: test/CMakeLists.txt (hunk 2 above)
Comment (bitoffdev) [reply test/CMakeLists.txt:6]: Yup! It's defined by cmake. We should be able to build Windows from Ubuntu. I'll put that on the backlog.

File: docs/WINDOWS.md (hunk 4 above)
Comment (bitoffdev) [reply docs/WINDOWS.md:71]: lol

File: docs/WINDOWS.md (hunk 1 above)
Comment (bitoffdev) [reply docs/WINDOWS.md:25]: That's fair. I'll put it on our backlog.

File: test/CMakeLists.txt (hunk 2 above)
Comment (bitoffdev) [reply test/CMakeLists.txt:6]: https://github.com/Nitepone/tetris-mint/issues/49

File: docs/WINDOWS.md (hunk 1 above)
Comment (bitoffdev) [reply docs/WINDOWS.md:25]: https://github.com/Nitepone/tetris-mint/issues/50

**Pull request merged**